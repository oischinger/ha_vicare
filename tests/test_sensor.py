"""Test ViCare sensors."""

from unittest.mock import MagicMock

import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.core import HomeAssistant


@pytest.mark.freeze_time("2022-04-19 07:53:05")
@pytest.mark.parametrize(
    "entity_id",
    [
        "sensor.vicare_outside_temperature",
        "sensor.vicare_boiler_temperature",
        "sensor.vicare_hot_water_max_temperature",
        "sensor.vicare_hot_water_min_temperature",
        "sensor.vicare_burner_hours",
        "sensor.vicare_burner_modulation",
        "sensor.vicare_burner_starts",
        "sensor.vicare_energy_consumption_this_month",
        "sensor.vicare_energy_consumption_this_year",
        "sensor.vicare_energy_consumption_today",
        "sensor.vicare_heating_gas_consumption_this_month",
        "sensor.vicare_heating_gas_consumption_this_week",
        "sensor.vicare_heating_gas_consumption_this_year",
        "sensor.vicare_heating_gas_consumption_today",
        "sensor.vicare_hot_water_gas_consumption_this_month",
        "sensor.vicare_hot_water_gas_consumption_this_week",
        "sensor.vicare_hot_water_gas_consumption_this_year",
        "sensor.vicare_hot_water_gas_consumption_today",
        "sensor.vicare_power_consumption_this_week",
    ],
)
async def test_gas_boiler_sensors(
    hass: HomeAssistant,
    mock_vicare_gas_boiler: MagicMock,
    snapshot: SnapshotAssertion,
    entity_id: str,
) -> None:
    """Test the ViCare Gas Boiler sensors."""
    state = hass.states.get(entity_id)
    assert state == snapshot


@pytest.mark.freeze_time("2022-04-19 07:53:05")
@pytest.mark.parametrize(
    "entity_id",
    [
        "sensor.vicare_energy_consumption_of_heating_current_day",
        "sensor.vicare_energy_consumption_of_heating_current_week",
        "sensor.vicare_energy_consumption_of_heating_current_month",
        "sensor.vicare_energy_consumption_of_heating_current_year",
        "sensor.vicare_energy_consumption_of_hot_water_heating_current_day",
        "sensor.vicare_energy_consumption_of_hot_water_heating_current_week",
        "sensor.vicare_energy_consumption_of_hot_water_heating_current_month",
        "sensor.vicare_energy_consumption_of_hot_water_heating_current_year",
    ],
)
async def test_heatpump_sensors(
    hass: HomeAssistant,
    mock_vicare_heatpump: MagicMock,
    snapshot: SnapshotAssertion,
    entity_id: str,
) -> None:
    """Test the ViCare Heatpump sensors."""
    state = hass.states.get(entity_id)
    assert state == snapshot


@pytest.mark.freeze_time("2022-04-19 07:53:05")
@pytest.mark.parametrize(
    "entity_id",
    [
        "sensor.vicare_room_temperature",
        "sensor.vicare_room_humidity",
    ],
)
async def test_room_sensor_sensors(
    hass: HomeAssistant,
    mock_vicare_room_sensor: MagicMock,
    snapshot: SnapshotAssertion,
    entity_id: str,
) -> None:
    """Test the ViCare Room Sensor sensors."""
    state = hass.states.get(entity_id)
    assert state == snapshot
