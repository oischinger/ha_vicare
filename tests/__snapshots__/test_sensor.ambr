# serializer version: 1
# name: test_gas_boiler_sensors[sensor.vicare_boiler_temperature]
  <state sensor.vicare_boiler_temperature=63; state_class=measurement, unit_of_measurement=°C, device_class=temperature, friendly_name=ViCare Boiler Temperature @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_burner_hours]
  <state sensor.vicare_burner_hours=18726.3; state_class=total_increasing, unit_of_measurement=h, icon=mdi:counter, friendly_name=ViCare Burner Hours @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_burner_modulation]
  <state sensor.vicare_burner_modulation=0; state_class=measurement, unit_of_measurement=%, icon=mdi:percent, friendly_name=ViCare Burner Modulation @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_burner_starts]
  <state sensor.vicare_burner_starts=14315; state_class=total_increasing, icon=mdi:counter, friendly_name=ViCare Burner Starts @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_energy_consumption_this_month]
  <state sensor.vicare_energy_consumption_this_month=7.843; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Energy consumption this month @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_energy_consumption_this_year]
  <state sensor.vicare_energy_consumption_this_year=207.106; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Energy consumption this year @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_energy_consumption_today]
  <state sensor.vicare_energy_consumption_today=0.219; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Energy consumption today @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_heating_gas_consumption_this_month]
  <state sensor.vicare_heating_gas_consumption_this_month=0; state_class=total_increasing, friendly_name=ViCare Heating gas consumption this month @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_heating_gas_consumption_this_week]
  <state sensor.vicare_heating_gas_consumption_this_week=0; state_class=total_increasing, friendly_name=ViCare Heating gas consumption this week @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_heating_gas_consumption_this_year]
  <state sensor.vicare_heating_gas_consumption_this_year=30946; state_class=total_increasing, friendly_name=ViCare Heating gas consumption this year @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_heating_gas_consumption_today]
  <state sensor.vicare_heating_gas_consumption_today=0; state_class=total_increasing, friendly_name=ViCare Heating gas consumption today @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_hot_water_gas_consumption_this_month]
  <state sensor.vicare_hot_water_gas_consumption_this_month=805; state_class=total_increasing, friendly_name=ViCare Hot water gas consumption this month @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_hot_water_gas_consumption_this_week]
  <state sensor.vicare_hot_water_gas_consumption_this_week=84; state_class=total_increasing, friendly_name=ViCare Hot water gas consumption this week @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_hot_water_gas_consumption_this_year]
  <state sensor.vicare_hot_water_gas_consumption_this_year=8203; state_class=total_increasing, friendly_name=ViCare Hot water gas consumption this year @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_hot_water_gas_consumption_today]
  <state sensor.vicare_hot_water_gas_consumption_today=22; state_class=total_increasing, friendly_name=ViCare Hot water gas consumption today @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_hot_water_max_temperature]
  <state sensor.vicare_hot_water_max_temperature=60; state_class=measurement, unit_of_measurement=°C, device_class=temperature, friendly_name=ViCare Hot Water Max Temperature @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_hot_water_min_temperature]
  <state sensor.vicare_hot_water_min_temperature=10; state_class=measurement, unit_of_measurement=°C, device_class=temperature, friendly_name=ViCare Hot Water Min Temperature @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_outside_temperature]
  <state sensor.vicare_outside_temperature=20.8; state_class=measurement, unit_of_measurement=°C, device_class=temperature, friendly_name=ViCare Outside Temperature @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_gas_boiler_sensors[sensor.vicare_power_consumption_this_week]
  <state sensor.vicare_power_consumption_this_week=0.829; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Power consumption this week @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_heatpump_sensors[sensor.vicare_energy_consumption_of_heating_current_day]
  <state sensor.vicare_energy_consumption_of_heating_current_day=6.3; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Energy consumption of heating current day @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_heatpump_sensors[sensor.vicare_energy_consumption_of_heating_current_month]
  <state sensor.vicare_energy_consumption_of_heating_current_month=152.4; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Energy consumption of heating current month @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_heatpump_sensors[sensor.vicare_energy_consumption_of_heating_current_week]
  None
# ---
# name: test_heatpump_sensors[sensor.vicare_energy_consumption_of_heating_current_year]
  <state sensor.vicare_energy_consumption_of_heating_current_year=831; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Energy consumption of heating current year @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_heatpump_sensors[sensor.vicare_energy_consumption_of_hot_water_heating_current_day]
  <state sensor.vicare_energy_consumption_of_hot_water_heating_current_day=2.2; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Energy consumption of hot water heating current day @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_heatpump_sensors[sensor.vicare_energy_consumption_of_hot_water_heating_current_month]
  <state sensor.vicare_energy_consumption_of_hot_water_heating_current_month=37.4; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Energy consumption of hot water heating current month @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_heatpump_sensors[sensor.vicare_energy_consumption_of_hot_water_heating_current_week]
  None
# ---
# name: test_heatpump_sensors[sensor.vicare_energy_consumption_of_hot_water_heating_current_year]
  <state sensor.vicare_energy_consumption_of_hot_water_heating_current_year=180.3; state_class=total_increasing, unit_of_measurement=kWh, device_class=energy, friendly_name=ViCare Energy consumption of hot water heating current year @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_room_sensor_sensors[sensor.vicare_room_humidity]
  <state sensor.vicare_room_humidity=56; state_class=measurement, unit_of_measurement=%, icon=mdi:percent, friendly_name=ViCare Room Humidity @ 2022-04-19T00:53:05-07:00>
# ---
# name: test_room_sensor_sensors[sensor.vicare_room_temperature]
  <state sensor.vicare_room_temperature=19.7; state_class=measurement, unit_of_measurement=°C, device_class=temperature, friendly_name=ViCare Room Temperature @ 2022-04-19T00:53:05-07:00>
# ---
